### PR TITLE
Generate helpful errors from dotenv! macro

### PR DESCRIPTION
**Pull request by Plecra**
An implementation for #9

### CODE REVIEW

**Comment by Plecra** [comment]
What would you like me to do about those regressions? I'm pretty sure I haven't removed any tests, and the only changes I made were in the macro implementation

**Comment by ZoeyR** [comment]
I'll look into it, but its been long enough since code coverage was run, I wouldn't be surprised if its just something going wrong with the codecov tool

**Comment by Plecra** [comment]
Ooh, I can actually understand the report now. It's identified the couple error cases that I introduced but didn't test. I don't think it's worth putting together a compile error test harness for those couple cases though...

**Comment by ghost** [comment]
@Plecra its also in codegen, which can't really be code coverage tested.

File: dotenv_codegen_implementation/src/lib.rs [hunk 1]
```diff
@@ -7,41 +7,50 @@ use proc_macro_hack::proc_macro_hack;
...
+            "expected a maximum of 2 arguments",
+        ));
     }
 
-    match env::var(var_name) {
-        Ok(val) => quote!(#val).into(),
-        Err(VarError::NotPresent) | Err(VarError::NotUnicode(_)) => panic!("{}", err_msg),
+    match env::var(var_name.value()) {
+        Ok(val) => Ok(quote!(#val).into()),
+        Err(VarError::NotPresent) | Err(VarError::NotUnicode(_)) => Err(syn::Error::new(
```
Comment (ghost) [inline dotenv_codegen_implementation/src/lib.rs]: I think we should break up this match arm now that we're producing more descriptive error messages. `NotUnicode` doesn't match with `environment variable {} not defined`

Comment (Plecra) [reply dotenv_codegen_implementation/src/lib.rs]: Taking a second look at this, I want to mirror the behavior of [`std::env!`](https://github.com/rust-lang/rust/blob/1033351a51dd3ca342a83d4be13f7554f0b4fb1e/src/librustc_builtin_macros/env.rs#L51-L93), which actually doesn't differentiate between the two errors. Not that that means we can't though

**Comment by Plecra** [comment]
Continued in #58

**Pull request closed without merging**